core/src/main/scala/org/apache/spark/Partitioner.scala

```diff
@@ -118,7 +118,8 @@ class RangePartitioner[K : Ordering : ClassTag, V](
       Array.empty
     } else {
       // This is the sample size we need to have roughly balanced output partitions, capped at 1M.
-      val sampleSize = math.min(20.0 * partitions, 1e6)
+      val maxSamples = rdd.sparkContext.getConf.getDouble("spark.partitioner.max_samples", 1e6)
```

Comment:
`max_samples` -> `maxSamples`. You are trying to turn *down* the max right? Just wondering out loud -- was 1e6 arbitrary and probably too big as a max to begin with? The max would only matter at 50000 partitions, which is huge. Would a max of... 1e5, 1e4 be more sensible for 99% of cases? I realize there's no single value that always works but wondering if we can avoid another flag.